include/mbedtls/platform.h

```diff
@@ -58,7 +58,7 @@ extern "C" {
 #endif
 #endif
 #if !defined(MBEDTLS_PLATFORM_STD_VSNPRINTF)
-#if defined(_WIN32)
+#if _WIN32_WINNT < 0x0A00 /* < Windows 10 */
```

Comment:
This test is not correct, it also catches Unix platforms: undefined macros evaluate to `0` according to the standard (I also confirmed it experimentally by adding a `#warning` inside that block: it get picked on Linux). You should first check if the symbol is defined before checking its value.

Also, the previous test was `defined(_WIN32)` and this one moves to `_WIN32_WINNT`: are there any circumstances when one it defined but not the other? If so, the test should for that.